interfaces/repo.go

```diff
@@ -716,6 +716,17 @@ func (r *Repository) DisconnectSnap(snapName string) ([]string, error) {
 	return result, nil
 }
 
+// isSpecialAutoConnectSnap checks special Name/Developer combinations to see
+// if this particular snap's connections should be automatically connected even
+// if the interfaces are not autoconnect and the snap is not an OS snap.
+// FIXME: remove once we have assertions that provide this feature
+func isSpecialAutoConnectSnap(snap *snap.Info) bool {
+	if snap.Name() == "canonical-livepatch" && snap.Developer == "canonical" {
+		return true
+	}
+	return false
+}
+
 // AutoConnectCandidates finds and returns viable auto-connection candidates
 // for a given plug.
 func (r *Repository) AutoConnectCandidates(plugSnapName, plugName string) []*Slot {
```

Comment:
The auto-connect logic is designed to detect an interface that was not connected before, that normally auto-connects, as a way to know about interfaces that were explicitly disconnected by the user.

When an interface auto-connectivity changes we treat that as if the user had disconnected that before because we don't maintain the list of explicitly made disconnect requests.

I wouldn't do anything about this in here.